tools/bowtie2.wdl

```diff
@@ -0,0 +1,410 @@
+### (https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#setting-function-options)
+struct Bowtie2Function {
+    String function_type
+    Float constant
+    Float coefficient
+}
+
+task build {
+    meta {
+        description: "Builds a Bowtie2 index from a FASTA file"
+        outputs: {
+            index_files: "Array of Bowtie2 index files",
+            bowtie_db_tar_gz: "Bowtie2 index files in tarred, gzipped format",
+        }
+    }
+
+    parameter_meta {
+        reference: "Reference genome in FASTA format (optionally gzipped)"
+        prefix: "Prefix for the Bowtie2 index files"
+        ncpu: "Number of threads to use"
+    }
+
+    input {
+        File reference
+        String prefix
+        Int ncpu = 4
+    }
+
+    String base = basename(reference, ".gz")
+
+    command <<<
+        set -euo pipefail
+
+        gunzip -c ~{reference} > ~{base} \
+           || ln -sf ~{reference} ~{base}
+
+        bowtie2-build --threads=~{ncpu} ~{base} ~{prefix}
+
+        tar -czf ~{prefix}.tar.gz ~{prefix}.*
+    >>>
+
+    output {
+        Array[File] index_files = glob("*.bt2")
+        File bowtie_db_tar_gz = prefix + ".tar.gz"
+    }
+
+    runtime {
+        cpu: ncpu
+        memory: "20 GB"
+        container: "ghcr.io/stjudecloud/bowtie2:2.5.4"
+        maxRetries: 1
+    }
+}
+
+task align {
+    meta {
+        description: "Aligns reads to a reference genome using Bowtie2"
+        outputs: {
+            aligned_bam: "Aligned reads in BAM format",
+            unpaired_unaligned: "Unpaired reads that didn't align",
+            unpaired_aligned: "Unpaired reads that aligned at least once",
+            paired_discordant: "Pairs that didn't align concordantly",
+            paired_concordant: "Pairs that aligned concordantly at least once",
+            metrics: "Metrics file",
+        }
+    }
+
+    parameter_meta {
+        bowtie_db_tar_gz: "Bowtie2 index files in tarred, gzipped format"
+        read_one_fastq_gz: "A gzipped FASTQ file containing read one information"
+        read_two_fastq_gz: "A gzipped FASTQ file containing read two information"
+        skip: "Skip the first N reads/read pairs"
+        upto: "Only align the first N reads/read pairs"
+        trim5: "Trim N bases from the 5' end of each read before alignment"
+        trim3: "Trim N bases from the 3' end of each read before alignment"
+        seed_mismatch: {
+            description: "Maximum number of mismatches in the seed [0, 1] during multiseed alignment.",
+            external_help: "https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic",
+            bowtie_option: "-N",
+        }
+        seed_substring: {
+            description: "Length of seed substring. >3, <32",
+            external_help: "https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#multiseed-heuristic",
+            bowtie_option: "-L",
+        }
+        dpad: "Include N extra reference characters on sides of dynamic programming table"
```

Comment:
This is unclear to me. At least this needs an `external_help` link.

Reply:
This is the entire entry from the [manual](https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml).
```
--dpad <int>   "Pads" dynamic programming problems by <int> columns on either side to allow gaps. Default: 15.
```

Reply:
Well that's not very illuminating... This seems like something you need intimate knowledge of the bowtie algorithm to tweak, and I clearly don't have that knowledge...

Reply:
![image](https://github.com/user-attachments/assets/990160ab-7a3c-4720-ad3c-f33c679669d9)
I'm not sure if you're familiar with dynamic programming for sequence alignment. You can visualize it as a matrix. You've got one sequence on the X-axis and another on the Y-axis. You're looking for the cheapest path from top-left to bottom-right.

I haven't thought about the algorithm in a long time, but with the description, you're essentially adding additional reference characters to the matrix, which allows the algorithm to open larger gaps if a better alignment can be found. So you'd get better at detecting long INDELs (I think). A more common operation would like be to set `dpad = 0` to find exact gapless matches.

Reply:
That explanation makes sense. Could you add a sentence to this description with that information? Like:
> `Set dpad=0 to find exact gapless matches`